src/Microsoft.IdentityModel.Tokens/Json/EncodedJsonWebKeyParameterNames.cs

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Encodings.Web;
+using System.Text.Json;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// JsonEncodedText used with utf8Writer when writing property names for JsonWebKey.
+    /// Common names are initialized on startup, lazy for others.
+    /// </summary>
+    internal static class EncodedJsonWebKeyParameterNames
+    {
+        internal static bool _algSet;
+        internal static JsonEncodedText _alg;
+        internal static bool _crvSet;
+        internal static JsonEncodedText _crv;
+        internal static bool _dSet;
+        internal static JsonEncodedText _d;
+        internal static bool _dpSet;
+        internal static JsonEncodedText _dp;
+        internal static bool _dqSet;
+        internal static JsonEncodedText _dq;
+        internal static bool _kSet;
+        internal static JsonEncodedText _k;
+        internal static bool _keyopsSet;
+        internal static JsonEncodedText _keyOps;
+        internal static bool _othSet;
+        internal static JsonEncodedText _oth;
+        internal static bool _pSet;
+        internal static JsonEncodedText _p;
+        internal static bool _qSet;
+        internal static JsonEncodedText _q;
+        internal static bool _qiSet;
+        internal static JsonEncodedText _qi;
+        internal static bool _x5tS256Set;
+        internal static JsonEncodedText _x5tS256;
+        internal static bool _x5uSet;
+        internal static JsonEncodedText _x5u;
+        internal static bool _xSet;
+        internal static JsonEncodedText _x;
+        internal static bool _ySet;
+        internal static JsonEncodedText _y;
+
+        public static JsonEncodedText Alg
+        {
+            get
+            {
+                if (!_algSet)
```

Comment:
I don't understand, why the lazy initialization? Why not just set all of these up directly?